tket/src/Mapping/include/Mapping/LexiRoute.hpp

```diff
@@ -0,0 +1,205 @@
+   * are  complete. If no swap is better, update return object to reflect this.
+   * @param swap Pair of Node comprising SWAP for checking
+   * @param lookahead Number of steps of lookahead emplyed for comparison
+   * @return Pair of bool, where true implies BRIDGE to be added
+   */
+  std::pair<bool, bool> check_bridge(
+      const std::pair<Node, Node>& swap, unsigned lookahead);
+
+  /**
+   * Returns a pair of distances, where the distances are between n1 & p1, and
+   * n2 & p2. Pair object is ordered such that the greatest distance is first.
+   *
+   * @param p0_first First Node in first interaction to find distance between
+   * @param p0_second Second Node in first interaction to find distance between
+   * @param p1_first First Node in second interaction to find distance between
+   * @param p1_second Second Node in second interaction to find distance between
+   * @return Pair of size_t, being distances on architecture graph
+   */
+  const std::pair<size_t, size_t> pair_distances(
+      const Node& p0_first, const Node& p0_second, const Node& p1_first,
+      const Node& p1_second) const;
+
+  /**
+   * It is always expected that at least one Node in a SWAP will be in some
+   * interaction. This method checks that the given swap will strictly decrease
+   * the distance for this interaction, and removes it from the swaps set if
+   * not.
+   *
+   * @param swaps Potential swaps to remove from
+   */
+  void remove_swaps_decreasing(swap_set_t& swaps);
+
+  // Architecture all new physical operations must respect
+  ArchitecturePtr architecture_;
+  //   Contains circuit for finding SWAP from and non-routed/routed boundary
+  std::shared_ptr<MappingFrontier>& mapping_frontier_;
+  //   Map between UnitID and UnitID they interact with at boundary
+  unit_map_t interacting_uids_;
+  //   Map between original circuit UnitID and new UnitID due to dynamic
+  //   placement
+  unit_map_t labelling_;
+  //   Set tracking which Architecture Node are present in Circuit
+  std::set<Node> assigned_nodes_;
+};
+
+// Child class of RoutingMethod, with overloaded methods for routing
+// MappingFrontier objects
+class LexiRouteRoutingMethod : public RoutingMethod {
+ public:
+  /**
+   * Checking and Routing methods redefined using LexiRoute. Only circuit depth,
+   * corresponding to lookahead, is a required parameter.
+   *
+   * @param _max_depth Number of layers of gates checked inr outed subcircuit.
+   */
+  LexiRouteRoutingMethod(unsigned _max_depth = 10);
+
+  /**
+   * @return true if method can route subcircuit, false if not
+   */
+  bool check_method(
+      const std::shared_ptr<MappingFrontier>& /*mapping_frontier*/,
+      const ArchitecturePtr& /*architecture*/) const override;
+
+  /**
+   * @param mapping_frontier Contains boundary of routed/unrouted circuit for
+   * modifying
+   * @param architecture Architecture providing physical constraints
+   * @return Logical to Physical mapping at boundary due to modification.
```

Comment:
Returning {} is correct - this method modifies the `mapping_frontier` object directly. Despite this being the case for `MultiGateReorder` as well, I think its worth keeping this return type as for some methods i suspect this will be easier. Ive updated the doc.